django_future_tasks/management/commands/populate_periodic_future_tasks.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import signal
+import time
+
+from django import db
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from croniter import croniter_range
+
+from django_future_tasks.models import FutureTask, PeriodicFutureTask
+
+logger = logging.getLogger("process_future_tasks")
+
+class Command(BaseCommand):
+    help = "Create single future tasks based on periodic future tasks from database"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--onetimerun",
+            action="append",
+            type=bool,
+            default=False,
+            help="Run command only one time",
+        )
+
+    def _handle_termination(self, *args, **kwargs):
+        self._running = False
+
+    def _handle_options(self, options):
+        self.tick = 1
+        self.one_time_run = options["onetimerun"]
+
+    @staticmethod
+    def periodic_tasks_for_processing():
+        return PeriodicFutureTask.objects.filter(
+            is_active=True
+        )
+
+    @staticmethod
+    def _convert_exception_args(args):
+        return [str(arg) for arg in args]
+
+    def handle_tick(self):
+        now = timezone.now()
+        periodic_task_list = self.periodic_tasks_for_processing()
+        logger.debug("Got {} periodic tasks for processing".format(len(periodic_task_list)))
+
+        for p_task in periodic_task_list:
+            last_population = p_task.last_task_creation
+            relevant_dts = croniter_range(last_population, now, p_task.cron_string)
+
+            for dt in relevant_dts:
```

Comment:
Please remove the loop. Currently this will create many tasks if the `populate_periodic_future_tasks` was not running.

Reply:
Yes, that was on purpose...

New suggestion pushed! (`last_task_creation` is set to `now` when starting `populate_periodic_future_tasks`.)